File: assets/samples/domain.availability/password/php/domain.avail.php
<?php

$domain = 'example.com';
$password = 'your whois api password';
$username = 'your whois api username';

$url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?'
     . "domainName={$domain}&username={$username}&password={$password}"
     . "&cmd=GET_DN_AVAILABILITY";

print(file_get_contents($url));

File: assets/samples/domain.availability/password/ruby/domain.avail.rb
require 'open-uri'

domain = 'example.com'
password = 'Your whois api password'
username = 'Your whois api username'

url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?'\
    + 'domainName=' + domain + '&username=' +username + '&password=' +password\
    + "&cmd=GET_DN_AVAILABILITY" + "&getMode=DNS_AND_WHOIS"

puts open(url).read